File: week-6/go/membersys/repository/memory_message_repository.go
package repository

import (
	. "Project/membersys/core"
)

type MemoryMessageRepository struct {
	memberRepository *MemoryMemberRepository
	db               []Message
}

func (r *MemoryMessageRepository) AddMessage(memberId int, content string) {
	member := r.memberRepository.GetMemberById(memberId)
	r.db = append(r.db, Message{Name: member.Name, Content: content})
}

func (r *MemoryMessageRepository) GetMessages() []Message {
	return r.db
}

File: week-7/py/member_system/repository/memory_repository/memory_message_repository.py
from typing import List
from member_system.core import Message
from member_system.repository.unit_of_work import MessageRepository
from member_system.repository.memory_repository.memory_member_repository import MemoryMemberRepository

class MemoryMessageRepository(MessageRepository):
    def __init__(self, memberRepository: MemoryMemberRepository):
        self.memberRepository = memberRepository
        self.__db: List[Message] = []

    def addMessage(self, __memberId: int, __content: str) -> None:
        member = self.memberRepository.getMemberById(__memberId)
        self.__db.append(Message(member.name, __content))
    
    def getMessages(self) -> List[Message]:
        return self.__db
